### PR TITLE
[ALLUXIO-2898] Make copyFromLocal use round robin by default

### PR DESCRIPTION
**Pull request by bf8086**
https://alluxio.atlassian.net/browse/ALLUXIO-2898

Added a configuration to determine the write location policy for copyFromLocal. Made it use round robin by default.

### CODE REVIEW

File: shell/src/main/java/alluxio/shell/command/CpCommand.java [hunk 1]
```diff
@@ -385,7 +390,17 @@ private void copyPath(AlluxioURI srcPath, AlluxioURI dstPath) throws AlluxioExce
...
       try (Closer closer = Closer.create()) {
-        os = closer.register(mFileSystem.createFile(dstPath));
+        FileWriteLocationPolicy locationPolicy;
+        try {
+          locationPolicy =
+              CommonUtils.createNewClassInstance(Configuration.<FileWriteLocationPolicy>getClass(
+                  PropertyKey.USER_FILE_COPY_FROM_LOCAL_WRITE_LOCATION_POLICY), new Class[] {},
+                  new Object[] {});
+        } catch (Exception e) {
+          throw Throwables.propagate(e);
```
Comment (apc999) [inline shell/src/main/java/alluxio/shell/command/CpCommand.java]: Just for your info, `Throwables.propagate(e)` has been depreciated by Google. Can you change this to be
```java
throw new RuntimeException(e);
```

File: core/common/src/main/java/alluxio/PropertyKey.java [hunk 2]
```diff
@@ -501,6 +501,9 @@
       create(Name.USER_FILE_BUFFER_BYTES, "8MB");
   public static final PropertyKey USER_FILE_CACHE_PARTIALLY_READ_BLOCK =
       create(Name.USER_FILE_CACHE_PARTIALLY_READ_BLOCK, true);
+  public static final PropertyKey USER_FILE_COPY_FROM_LOCAL_WRITE_LOCATION_POLICY =
```
Comment (yupeng9) [inline core/common/src/main/java/alluxio/PropertyKey.java]: Can you also update the [docs](http://www.alluxio.org/docs/master/en/Configuration-Settings.html) to include this configuration?

File: docs/en/Debugging-Guide.md [hunk 3]
```diff
@@ -140,7 +140,7 @@ Please read [Configuration-Settings](Configuration-Settings.html) for how to cus
 
 A: This error indicates insufficient space left on Alluxio workers to complete your write request.
 
-- If you are copying a file to Alluxio using `copyFromLocal`, by default this shell command applies `LocalFirstPolicy`
+- Before version 1.6.0, if you are copying a file to Alluxio using `copyFromLocal`, by default this shell command applies `LocalFirstPolicy`
```
Comment (yupeng9) [inline docs/en/Debugging-Guide.md]: Can you change the doc to reflect the lastest status?

**Comment by apc999** [comment]
LGTM per yupeng's comment

File: docs/en/Debugging-Guide.md [hunk 4]
```diff
@@ -140,14 +140,19 @@ Please read [Configuration-Settings](Configuration-Settings.html) for how to cus
 
 A: This error indicates insufficient space left on Alluxio workers to complete your write request.
 
-- Before version 1.6.0, if you are copying a file to Alluxio using `copyFromLocal`, by default this shell command applies `LocalFirstPolicy`
-and stores data on the local worker (see [location policy](File-System-API.html#location-policy)).
+- If you are copying a file to Alluxio using `copyFromLocal` with `LocalFirstPolicy`,
```
Comment (yupeng9) [inline docs/en/Debugging-Guide.md]: I suggest we talk about the default policy of the latest version at beginning, and move this description on potential error of `LocalFirstPolicy` to the paragraph of `before version 1.6.0`.

Comment (bf8086) [inline docs/en/Debugging-Guide.md]: Good suggestions.

**Pull request merged**